sdk/src/main/scala/com/horizen/api/http/SidechainBlockActor.scala

```diff
@@ -88,47 +87,17 @@ class SidechainBlockActor[PMOD <: PersistentNodeViewModifier, SI <: SidechainSyn
 
   }
 
-  // Note: It should be used only in regtest
-  protected def generateSidechainBlocks: Receive = {
-    case GenerateSidechainBlocks(blockCount) =>
-      // Try to forge blockCount blocks, collect their ids and wait for
-      var generatedIds: Seq[ModifierId] = Seq()
-      for (i <- 1 to blockCount) {
-        val future = forgerRef ? TryForgeNextBlock
-        Await.result(future, timeoutDuration).asInstanceOf[Try[ModifierId]] match {
-          case Success(id) =>
-            generatedIds = id +: generatedIds
-            if (i == blockCount) {
-              // Create a promise, that will wait for blocks applying result from Node
-              val prom = Promise[Try[Seq[ModifierId]]]()
-              generatedBlockGroups += (id -> generatedIds)
-              generatedBlocksPromises += (id -> prom)
-              sender() ! prom.future
-            }
-
-          case Failure(ex) =>
-            sender() ! Future[Try[Seq[ModifierId]]](Failure(ex))
-        }
-      }
-  }
-
-  protected def tryToSubmitBlock: Receive = {
-    case SubmitSidechainBlock(blockBytes: Array[Byte]) =>
-      val future = forgerRef ? TrySubmitBlock(blockBytes)
-      Await.result(future, timeoutDuration).asInstanceOf[Try[ModifierId]] match {
-        case Success(id) =>
-          // Create a promise, that will wait for block applying result from Node
-          val prom = Promise[Try[ModifierId]]()
-          submitBlockPromises += (id -> prom)
-          sender() ! prom.future
-
-        case Failure(ex) =>
-          sender() ! Future[Try[ModifierId]](Failure(ex))
+  protected def forgeNextBlock: Receive = {
+    case messageToForger @ TryForgeNextBlockForEpochAndSlot(epochNumber, slotNumber) =>
+    {
+        val future = forgerRef ? messageToForger
+        val forgerResult = Await.result(future, timeoutDuration).asInstanceOf[Try[ModifierId]]
+        sender() ! forgerResult
       }
   }
 
   override def receive: Receive = {
-    sidechainNodeViewHolderEvents orElse generateSidechainBlocks orElse tryToSubmitBlock orElse {
+    sidechainNodeViewHolderEvents orElse {
```

Comment:
add `forgeNextBlock` as well. I guess, that message sent by API is still not processed.